systest/testcontext/context.go

```diff
@@ -365,3 +375,12 @@ func New(t *testing.T, opts ...Opt) *Context {
 	cctx.Log.Infow("using", "namespace", cctx.Namespace)
 	return cctx
 }
+
+func (c *Context) CheckFail() error {
+	select {
+	case <-failed:
+		return errors.New("test suite failed. aborting test execution")
+	default:
+	}
+	return nil
+}
```

Comment:
Wouldn't `atomic.Bool` be more straightforward to use?

```go
failed atomic.Bool
...
t.Cleanup(func() {
	if t.Failed() {
		failed.Set(true)
	}
})
```

```suggestion
func (c *Context) CheckFail() error {
	if failed.Load() {
		return errors.New("test suite failed. aborting test execution")
	}
	return nil
}
```